tests/aws/services/lambda_/test_lambda_performance.py

```diff
@@ -115,11 +115,51 @@ def increment(self):
 
 
 @markers.aws.unknown
-def test_number_of_function_versions(create_lambda_function, s3_bucket, aws_client):
+def test_number_of_function_versions_sync(create_lambda_function, s3_bucket, aws_client):
     """Test how many function versions LocalStack can support.
-    This test mainly idles and does one sync + async invoke as a sanity check."""
+    This test performs one sync invoke for each function version."""
+    num_function_versions = 400
+
+    function_name = f"test-lambda-perf-{short_uid()}"
+    create_lambda_function(
+        handler_file=TEST_LAMBDA_PYTHON_S3_INTEGRATION,
+        func_name=function_name,
+        runtime=Runtime.python3_12,
+        Environment={"Variables": {"S3_BUCKET_NAME": s3_bucket}},
+    )
+
+    function_versions = []
+    for _ in range(num_function_versions):
+        response = aws_client.lambda_.publish_version(FunctionName=function_name)
+        function_versions.append(response["Version"])
+
+    # idle for a while to see if the pollers can manage the load and any errors occur
+    time.sleep(20)
+
+    LOG.info("Invoke each function once synchronously")
+    request_ids = set()
+    for fn_version in function_versions:
+        result_sync = aws_client.lambda_.invoke(
+            FunctionName=function_name,
+            InvocationType=InvocationType.RequestResponse,
+            Qualifier=fn_version,
+        )
+        assert "FunctionError" not in result_sync
+        request_id = result_sync["ResponseMetadata"]["RequestId"]
+        request_ids.add(request_id)
+        assert result_sync["ExecutedVersion"] == fn_version
+
+    LOG.info("Validate invocations")
+    s3_request_ids = get_s3_keys(aws_client, s3_bucket)
+    assert set(s3_request_ids) == request_ids
+
+
+@markers.aws.unknown
```

Comment:
I guess it depends:
* For a LocalStack-specific parametrization yes, we should guard it because we don't want to execute this against AWS 🤑 
* For the test itself, I think we should target it to be aws_validated whenever possible. For example, it prevents us from taking unintended shortcuts due to missing AWS parity.

We could adjust the concurrency like `3 if is_aws_cloud() else N`.